src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+				String legacyID = record.get(COSMIC_LEGACY_MUTATION_ID);
+				String mutationID = record.get(COSMIC_MUTATION_ID);
+				String genomicID = record.get(COSMIC_GENOMIC_MUTATION_ID);
+				if (updaters.containsKey(legacyID))
+				{
+					updaters.get(legacyID).setValid(true); // only VALID if in MutantExport...
+					updaters.get(legacyID).getMutationIDs().add(mutationID);
+					updaters.get(legacyID).setCosvIdentifier(genomicID);
+				}
+			});
+		}
+	}
+
+	/**
+	 * Determines the prefixes for COSMIC identifiers. The rule is:
+	 * IF an object has EWASes and there is an EWAS with a FragmentReplacedModification or a FragmentInsertionModification whose referenceSequence 
+	 * is NOT the referenceEntity of the EWAS... then the suggested prefix will be COSF (for Fusion), otherwise, COSM is suggested.
+	 * @param cosmicObjects Objects that are identified by a COSMIC identifier.
+	 * @return A map of <code>COSMICIdentifierUpdater</code>, keyed by COSMIC identifier.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws IOException
+	 */
+	static Map<String, COSMICIdentifierUpdater> determinePrefixes(Collection<GKInstance> cosmicObjects) throws InvalidAttributeException, Exception, IOException
+	{
+		Map<String, COSMICIdentifierUpdater> updates = new HashMap<>();
+		
+		// Create the reports directory if it's missing.
+		if (!Files.exists(Paths.get(COSMICUpdateUtil.reportsDirectoryPath)))
+		{
+			Files.createDirectories(Paths.get(COSMICUpdateUtil.reportsDirectoryPath));
+		}
+		
+		try(CSVPrinter nonEWASPrinter = new CSVPrinter(new FileWriter(COSMICUpdateUtil.reportsDirectoryPath + "/nonEWASObjectsWithCOSMICIdentifiers_"+dateSuffix+".csv"), CSVFormat.DEFAULT.withHeader("COSMIC identifier", "Referred-to-by CadidateSet (or other non-EWAS)"));
+			CSVPrinter identifiersWithNoReferrerPrinter = new CSVPrinter(new FileWriter(COSMICUpdateUtil.reportsDirectoryPath + "/COSMICIdentifiersNoReferrers_"+dateSuffix+".csv"), CSVFormat.DEFAULT.withHeader("COSMIC identifier")))
+		{
+			for (GKInstance inst : cosmicObjects)
+			{
+				String identifier = (String)inst.getAttributeValue(ReactomeJavaConstants.identifier);
+			
+				COSMICIdentifierUpdater updater = new COSMICIdentifierUpdater();
+				updater.setIdentifier(identifier);
+				updater.setDbID(inst.getDBID());
+
+				@SuppressWarnings("unchecked")
+				Collection<GKInstance> EWASes = inst.getReferers(ReactomeJavaConstants.crossReference);
+				// If NO EWASes exist, then log this information.
+				if (EWASes == null || EWASes.isEmpty())
+				{
+					identifiersWithNoReferrerPrinter.printRecord(identifier);
+				}
+				else
+				{
+					// Check the EWASes for mismatches between referenceSequence identifier and main identifier.
+					// If there's a mismatch then suggest COSF, else suggest COSM.
+					checkEWASes(nonEWASPrinter, identifier, updater, EWASes);
+				}
+				
+				// If the identifier starts with C it's not a numeric identifier.
+				if (identifier.startsWith("C"))
```

Comment:
Same as https://github.com/reactome/release-cosmic-update/pull/1/files#r552224756 - the check itself is for the identifier starting with 'C' (which is obviously valid), but the comment is regarding it not being numeric

Reply:
Checking that the first character is numeric would require a regex (or parsing a string to numeric) vs. checking that the first character is "C". Checking the first character of a string is much simpler than evaluating a regular expression or parsing a string to a number (which could also involve regular expressions).

Reply:
Similar to https://github.com/reactome/release-cosmic-update/pull/1/files#r568852047